enhancements/machine-config/manage-boot-images.md

```diff
@@ -319,6 +324,194 @@ When [MachineDeployments](https://cluster-api.sigs.k8s.io/developer/architecture
+- Marketplace - Azure Marketplace published images (paid offerings with OCP, OPP, OKE variants)
+
+**OKD Streams:**
+- SCOS-OKD - Singular SCOS stream on AWS
+
+The MCO examines the current MachineSet boot image, determines its stream, and then determines the desired image from the appropriate stream metadata. Once all known streams are handled, the MCO should enter a degraded state if it expects to manage a boot image but cannot determine its stream.
+
+#### Platform-Specific Stream Detection and Update Strategy
+
+##### AWS
+
+###### History Bootimage Tracking (Existing Clusters)
+
+For existing upgrade clusters, determining the boot image source requires examining multiple data sources:
+
+**AWS MachineSet backed installer-shipped image:**
+- Create a historical list of all installer-shipped AMIs from RHCOS.json files
+- Match current MachineSet AMI against this historical data
+
+**ROSA:**
+- Create a list of all ROSA AMIs based on GitLab cluster image set metadata (see Red Hat internal Gitlab instance - service/clusterimagesets)
+
+**AWS Marketplace:**
+Two potential approaches:
+1. Filter out all non-ROSA marketplace published images and curate a list
+2. Default to not supported - require user to update to a "new" marketplace image once skew enforcement is in place
+
+**Detection fallback:**
+If the boot image is not found in any historical list:
+- Use AWS SDK to check publisher and RHCOS version:
+  - **Deregistered/Not Found**: We will estimate the skew based on the install version of the cluster, and have the skew enforcement rules apply to it as normal
+  - **Marketplace-published**: Could be UPI or ROSA - apply normal skew enforcement rules, but require update to future images with proper metadata
+  - **RHCOS**: Follow regular skew rules (consider raising warning since it should be in the historical list)
+
+###### Future Bootimage Tracking (New Images)
+
+**Installer changes:**
+
+Create multiple JSON files of AMI variant metadata for different streams:
+- rhcos (standard IPI)
+- rosa
+- marketplace-ocp
+- marketplace-oke
+- marketplace-opp
+
+**Image creation:**
+
+- All variants are tagged with `variantType` metadata to enable deterministic stream detection
+- The tagging will be done with AWS's [tagSet](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Image.html) field, with `variantType as the Key and stream name as the value
+
+##### GCP
+
+###### History Bootimage Tracking (Existing Clusters)
+
+GCP boot images use format: `projects/<project-name>/global/images/<image-name>`, which we can use to parse the historical stream where possible.
+
+**GCP IPI installs:**
+- Installer images have project name `rhcos-cloud`
+
+**GCP Marketplace:**
+- Marketplace images have project name `redhat-marketplace-public`
+- The image name should contain the variant.
+
+**GCP UPI:**
+- UPI installations typically upload their own images
+- These fall into the non-managed case and require manual updates
```

Comment:
I think for GCP UPI, users for the most part would still be using the RHCOS published images (this would not be the case in Azure). Checking on this, I see that our [in-repo docs](https://github.com/openshift/installer/blob/main/docs/user/gcp/install_upi.md#create-the-cluster-image) are actually confusing: the section is called "Create the cluster image" but if you examine the commands being run, the image is just being pulled from the rhcos stream.

Users could indeed create their own images, but I suspect for most cases UPI will be the same as IPI.